sequencer/src/block.rs

```diff
@@ -17,28 +19,131 @@ use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use std::{
     env,
     fmt::{Debug, Display},
+    ops::Add,
     time::Duration,
 };
 use time::OffsetDateTime;
-use typenum::U3;
+use typenum::Unsigned;
+use typenum::U2;
 
 pub type BlockMerkleTree = LightWeightSHA3MerkleTree<Commitment<Header>>;
 pub type BlockMerkleCommitment = <BlockMerkleTree as MerkleTreeScheme>::Commitment;
 
-#[derive(Default)]
+// New Type for `U256` in order to implement `CanonicalSerialize` and
+// `CanonicalDeserialize`
+#[derive(Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct FeeElement(U256);
+// New Type for `Address` in order to implement `CanonicalSerialize` and
+// `CanonicalDeserialize`
+#[derive(
+    Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, PartialOrd, Ord,
+)]
+pub struct FeeIndex(Address);
+
+impl Add for FeeElement {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        Self(self.0.add(other.0))
+    }
+}
+
+impl Valid for FeeElement {
+    fn check(&self) -> Result<(), SerializationError> {
+        Ok(())
+    }
+}
+
+impl Valid for FeeIndex {
+    fn check(&self) -> Result<(), SerializationError> {
+        Ok(())
+    }
+}
+
+impl CanonicalSerialize for FeeElement {
+    fn serialize_with_mode<W: std::io::prelude::Write>(
+        &self,
+        mut writer: W,
+        _compress: Compress,
+    ) -> Result<(), SerializationError> {
+        let mut bytes = [0u8; core::mem::size_of::<U256>()];
+        self.0.to_little_endian(&mut bytes);
```

Comment:
Nit: maybe cleaner to do `let bytes = self.0.0`?

Reply:
I'm not sure if it is really much cleaner as as the internal type of `U256` is `&[u64: 4]`.

Reply:
Oh right. Never mind